astroquery/sdss/core.py

```diff
@@ -56,10 +58,78 @@ class SDSSClass(BaseQuery):
     TIMEOUT = conf.timeout
 
     QUERY_URL = 'http://skyserver.sdss3.org/public/en/tools/search/x_sql.aspx'
+    XID_URL = 'http://skyserver.sdss.org/dr12/en/tools/crossid/x_crossid.aspx'
+
+    def query_crossid_async(self, coordinates, obj_names=None,
+                            photoobj_fields=None, specobj_fields=None,
+                            get_query_payload=False, timeout=TIMEOUT,
+                            radius=u.degree / 1800, output_format='csv'):
```

Comment:
Yes, it was inherited from the other function :smile: 

 Also it may be more optimal to change ``_args_to_payload()`` to support building queries for multiple APIs, as it would remove some code duplication.
I'm short of time at the moment, just wanted to have a draft PR out while it's fresh (less likely that I forget to finish it once it's on github and not just a stale local branch).